client/quantum_serverless/core/job.py

```diff
@@ -131,21 +132,13 @@ def list(self, **kwargs) -> List["Job"]:
 
     def run(self, program: Program, arguments: Optional[Dict[str, Any]] = None):
         arguments = arguments or {}
-        arguments_string = ""
-        if arguments is not None:
-            arg_list = []
-            for key, value in arguments.items():
-                if isinstance(value, dict):
-                    arg_list.append(f"--{key}='{json.dumps(value)}'")
-                else:
-                    arg_list.append(f"--{key}={value}")
-            arguments_string = " ".join(arg_list)
-        entrypoint = f"python {program.entrypoint} {arguments_string}"
+        entrypoint = f"python {program.entrypoint}"
 
         # set program name so OT can use it as parent span name
         env_vars = {
             **(program.env_vars or {}),
             **{OT_PROGRAM_NAME: program.title},
+            **{ENV_JOB_ARGUMENTS: json.dumps(arguments, cls=QiskitObjectsEncoder)},
```

Comment:
non-blocking question: is this envvar currently being used? (the github search wasn't super helpful...)

Reply:
yes, in `get_arguments` function

Reply:
didn't phrase this well, but what I was wondering about if it's being used to set how the program runs ( I probably should've put this on the line where `entrypoint` was defined... ), but looking at it a bit more, it seems like this is just moving the ray client more in line with how the other clients take arguments

Reply:
exactly :)